### DISCUSSION TITLE
Experiencing Duplicate Notifications Issue Recently

### DISCUSSION THREAD

**Discussion by shucunbin** [relayrides/pushy#1045] [Q&A]
Hello,

Our app has been using Pushy for notifications without any issues for quite some time. However, starting around September this year, we've been receiving reports from users about receiving duplicate notifications.

After debugging, we've ensured that each notification only triggers Pushy once, and this issue is not easily reproducible. It might randomly occur every few days.

Initially, when this problem surfaced, we were using Pushy version 0.14.2. Recently, we upgraded to Pushy version 0.15.2, but the issue persists.

I suspect it might be related to iOS 17 since there seems to be a slight overlap in timing. Has anyone else encountered a similar problem? We sincerely hope to receive assistance from the community. Thank you for your help~

**Comment by yyDeveloper** [comment]
Our app has also encountered the same problem, but we still haven't found the cause. We hope to get help !

**Comment by jchambers** [comment]
Given the information available, I don't think this is a Pushy problem, and is likely an issue on Apple's end.

@shucunbin as you point out, it sounds like this started happening spontaneously, and is not tied to any change in Pushy itself. It also sounds like this behavior aligns with the release of iOS 17, and in my experience, iOS releases often align with upstream changes to the APNs servers (i.e. they roll out support for new features in the push notification servers at the same time as the OS that needs those features).

I've also seen complaints about double in various corners of the internet, and I strongly suspect that at least some of those cases come from apps that are not using Pushy to send notifications.

I think the best way to address this issue is to raise it with Apple directly. I acknowledge it's very difficult to obtain this kind of debugging information, but if you can get the APNs ID (a UUID) of a notification that got double-delivered, that can be a big help in tracking the problem down.

**Reply by shucunbin** [reply]
Thank you  for your response. Over the past month or two, I have been observing the push notifications of several apps I frequently use and have noticed a similar issue. I receive two (or more) identical push notifications almost simultaneously. However, some apps have implemented optimizations for this issue. Even though I receive two notifications (consecutive pop-ups  or sound alerts), only one is displayed in the iOS Notification Center.

Based on the above observations, I made the following adjustments when creating the **SimpleApnsPushNotification** instance:
1. Used a custom **apnsId** to ensure that the apnsId for each notification is unique. Unfortunately, this did not solve the problem.
2. Building on point 1, further utilized a custom collapseId to ensure that the **collapseId** for each notification is unique. This partially resolved the issue. While I still receive two duplicate notifications (consecutive pop-ups or consecutive sound alerts), only one is displayed in the iOS Notification Center. This might be the approach adopted by some apps currently.

**Comment by yyDeveloper** [comment]
谢谢。您的邮件已收到，我会尽快处理
